inginious/frontend/task_problems.py

```diff
@@ -185,12 +185,12 @@ def show_input(self, template_helper, language, seed):
                 if entry['valid'] and limit > 0:
                     choices.append(entry)
                     limit = limit - 1
-
-        rand.shuffle(choices)
-
+        if self._shuffle:
+            rand.shuffle(choices)
+        else:
+            choices = sorted(choices, key=lambda k: k['index'])
```

Comment:
You short-circuit only one shuffle here, and sort an already randomly generated list. Your list of choice will be displayed in the teacher order, but the choice set will still be randomly generated according to the seed.

I don't know if this is the expected result.

Reply:
As there is a pick of correct answer (multiple or not) just after the first shuffle, it ruins the order in any case. So i'm not sure that a course-circuit is necessary on first shuffle as the order will not be respected anyway.

Reply:
What would be the point of this feature then ?  The idea is to have the same rendering whatever the random seed parameters are. Currently, you will not have the same rendering if the language is different for instance.